kasa/tapoplug.py

```diff
@@ -0,0 +1,194 @@
+class TapoPlug(SmartPlug):
+    def __init__(self, host: str, *, port: Optional[int] = None, credentials) -> None:
+        # TODO: we are calling smartdevice here to avoid smartplug internal handling
+        SmartDevice.__init__(self, host, port=port, credentials=credentials)
+        # TODO: this is needed as we don't call smartplug ctor
+        self._device_type = DeviceType.Plug
+        self._tapo_client = None
+        self.protocol = TapoAESProtocol(host, credentials=credentials)
+        self._last_update = None
+
+    async def update(self, update_children: bool = True):
+        if self.credentials is None or self.credentials.username is None:
+            raise AuthenticationException("Tapo plug requires authentication.")
+
+        self._info = AttributeDict(await self.protocol.query("get_device_info"))
+
+        self._usage = AttributeDict(await self.protocol.query("get_device_usage"))
+        self._energy = AttributeDict(await self.protocol.query("get_energy_usage"))
+        self._emeter = AttributeDict(await self.protocol.query("get_current_power"))
+        self._time = AttributeDict(await self.protocol.query("get_device_time"))
+        self._last_update = self._data = {
+            "info": self._info,
+            "usage": self._usage,
+            "emeter": self._emeter,
+            "energy": self._energy,
+            "time": self._time,
+        }
+
+        _LOGGER.debug("Got an update: %s", self._data)
+
+    @property
+    def sys_info(self) -> Dict[str, Any]:
+        return self._info
+
+    @property
+    def model(self) -> str:
+        return self._info.model
+
+    @property
+    def alias(self) -> str:
+        return base64.b64decode(self._info.nickname).decode()
+
+    @property
+    def time(self) -> datetime:
+        """Disabled as 'region' is '' on my test device
+        File "/x/python-kasa/kasa/tapoplug.py", line 80, in time
+            return self._time.local_time()
+                   ^^^^^^^^^^^^^^^^^^^^^^^
+        File "x/plugp100/responses/time_info.py", line 19, in local_time
+            return datetime.fromtimestamp(self.timestamp, tz=ZoneInfo(self.region))
+                                                          ^^^^^^^^^^^^^^^^^^^^^
+        """
+        return None
+        return self._time.local_time()
+
+    @property
+    def timezone(self) -> Dict:
+        """Disabled as 'region' is '' on my test device
+        File "/x/python-kasa/kasa/tapoplug.py", line 80, in time
+            return self._time.local_time()
+                   ^^^^^^^^^^^^^^^^^^^^^^^
+        File "x/plugp100/responses/time_info.py", line 19, in local_time
+            return datetime.fromtimestamp(self.timestamp, tz=ZoneInfo(self.region))
+                                                          ^^^^^^^^^^^^^^^^^^^^^
+        """
+        return None
+        return {"timezone": self._info.timezone, "timediff": self._info.time_difference}
```

Comment:
I didn't really understand your comments on these properties.  Can you explain what is going on with this code?

Reply:
Ahh, sorry, looks like I missed adding the real exception cause to those... So basically, calling `zoneinfo.ZoneInfo()` with an empty string causes a `ValueError: ZoneInfo keys must be normalized relative paths, got:` being raised.

As my device does not return a valid timezone for some reason, I commented them out to avoid the cli tool bailing out when printing out the device status:
```
'time': {'time_diff': 60, 'timestamp': 1701123524, 'region': ''}
```
Now that this does not depend on plugp100, we could handle such empty locales (e.g., by defaulting to the local timezone?).

Reply:
Done